Sources_v3/Query/Pagination.swift

```diff
@@ -53,6 +53,12 @@ public struct MessagesPagination: Encodable, Equatable {
         self.parameter = parameter
     }
     
+    /// Initializer with required page size.
+    init(pageSize: Int, parameter: PaginationParameter? = nil) {
+        self.pageSize = pageSize
+        self.parameter = parameter
+    }
+    
```

Comment:
Well, doesn't this shows something is wrong with the intializer we have?

Reply:
I checked on that.
The reason for that failable initializer is creating a new channel with `ChannelEditDetailPayload` in `ChannelQuery` where you don't need pagination.
We can move handling in `ChannelQuery` but I don't think it's necessary and I'm fine with having 2 initializers.